src/probe/probe.c

```diff
@@ -200,14 +261,20 @@ static int probe_dma_init(struct probe_dma_ext *dma, uint32_t direction)
 static int probe_dma_deinit(struct probe_dma_ext *dma)
 {
 	int err = 0;
-
+#if CONFIG_ZEPHYR_NATIVE_DRIVERS
+	err = dma_stop(dma->dc.dmac->z_dev, dma->dc.chan->index);
+#else
 	err = dma_stop_legacy(dma->dc.chan);
+#endif
 	if (err < 0) {
 		tr_err(&pr_tr, "probe_dma_deinit(): dma_stop() failed");
 		return err;
 	}
-
+#if CONFIG_ZEPHYR_NATIVE_DRIVERS
+	dma_release_channel(dma->dc.dmac->z_dev, dma->dc.chan->index);
+#else
 	dma_channel_put_legacy(dma->dc.chan);
+#endif
```

Comment:
just a suggestion, use one #if from line 264 - 277, this can  remove one more "#if #else #endif", overhead is duplicated err message.